components/resource-creation/ResourceDisplay.tsx

```diff
@@ -84,6 +87,79 @@ function ResourceDisplay() {
     setSelectedDataRequirement({ name: '', content: null });
   };
 
+  const dateForResource = (resource: fhir4.FhirResource ) => {
+    if (!resource || !PrimaryDatePaths?.parsedPrimaryDatePaths) {
+      return 'N/A'
+    }
+
+    const dateinfo =  PrimaryDatePaths.parsedPrimaryDatePaths[resource.resourceType] 
+    if (!dateinfo) {
+      return 'N/A'
+    }
+
+    for (const nameOfResourceDate of Object.keys(dateinfo)) {
+      // If only one dataType
+      const resourceDateData = fhirpath.evaluate(resource, nameOfResourceDate)[0]
+      if(dateinfo[nameOfResourceDate].dataTypes.length == 1 && resourceDateData){
+        // If the only dataType is a period 
+        if(dateinfo[nameOfResourceDate].dataTypes[0] == 'Period'){
+          return formatPeriod(resource as fhir4.Period, nameOfResourceDate, nameOfResourceDate)
```

Comment:
```suggestion
          return formatPeriod(resource, nameOfResourceDate, nameOfResourceDate)
```
We don't need to cast this fhir4.FhirResource type to a fhir4.Period, just leave it as a fhir4.FhirResource and change the input type of resource in `formatPeriod`